### ISSUE TITLE
Label encoding can change if I'm not careful

### ISSUE THREAD

**Issue by ben-heil** [greenelab/saged#23]
Right now labels are encoded at dataset creation time using sklearn.LabelEncoder, which sorts the labels before encoding them [method one](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/preprocessing/_label.py#L37), [method 2](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/preprocessing/_label.py#L58). This means that for classification where the labels are the same between datasets, the labels' encodings should also be the same. However, if there are more labels in the training set than validation set, the validation might be comparing apples to oranges. 

This might be the cause of the bad performance in the `all_labels` analysis, and definitely is something to make sure to get right. I'll be adding the logic to account for this (`LabeledDataset.get_label_encoder()` and `LabeledDataset.set_label_encoder()`) to the next PR
